src/units/types.cpp

```diff
@@ -493,11 +493,43 @@ t_string unit_type::unit_description() const
 
 bool unit_type::has_special_notes() const
 {
-	return !special_notes_.empty();
+	if(!special_notes_.empty()) return true;
+	for(const config::any_child ability : abilities_cfg().all_children_range()) {
+		if(ability.cfg.has_attribute("special_note")) {
+			return true;
+		}
+	}
+	for(const auto& attack : attacks()) {
+		for(const config::any_child ability : attack.specials().all_children_range()) {
+			if(ability.cfg.has_attribute("special_note")) {
+				return true;
+			}
+		}
+	}
+	return false;
```

Comment:
I think the has_special_notes() function can simply be removed, as the only caller is the help_topic_generator, and if this returns true then it must call special_notes() anyway. So it could just call special_notes() and check whether the vector is empty.

My interest here is working out what to do with `NOTE_DEFENSE_CAP` - there's already logic in help_topic_generator.cpp to show the cap column if and only if it has data; that special note could be auto-added in the generator.

Reply:
This comment is as much notes to myself as a reply, still trying to work out what fits better.

At the moment the few translatable strings in src/units/*.cpp are all in textdomain `wesnoth`, while both special-notes.cfg and help_topic_generators.cpp are in `wesnoth-help`. I'd prefer to keep it in the same domain as the other notes.

For the current PR, there should be logic somewhere so that a unit with two arcane attacks only gets one copy of the note on its help page. In future, but not for this PR, it might be nice to have a way to hide some special notes that the user is familiar with. Those two bits of logic could fit together, OTOH they're probably trivial logic (if not trivial UI) anyway.

Reply:
> For the current PR, there should be logic somewhere so that a unit with two arcane attacks only gets one copy of the note on its help page.

What are you talking about? The only way you could get two copies of the note with the current PR is is you included both `{SPECIAL_NOTES_ARCANE}` and `{NOTE_ARCANE}`, which you're not supposed to do.

The logic you describe would only become important if the arcane note were automatically added.